monitoringlinter/README.md

```diff
@@ -0,0 +1,82 @@
+# monitoring-linter
+
+Monitoring Linter is a Golang linter designed for ensuring that in Kubernetes operators projects, 
+monitoring related practices are implemented within `pkg/monitoring` directory, using [operator-observability](https://github.com/machadovilaca/operator-observability/tree/main) methods.
+
+## Installation
+
+```shell
+go install github.com/kubevirt/monitoring/monitoringlinter/cmd/monitoringlinter@latest
+```
+
+## Usage
+Once installed, you can run the Monitoring Linter against your Kubernetes operator project by using the following command:
+
+```shell
+monitoringlinter  ./...
+```
+
+
+## Linter Rules
+Currently, the linter verifies that all metrics, alerts and recording rules registrations 
+are implemented within `pkg/monitoring` directory, using [operator-observability](https://github.com/machadovilaca/operator-observability/tree/main) package registration methods.
+[Prometheus registration methods](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#Registerer) are restricted across all the project.
+
+**Examples**:
+
+- The following code is restricted across all the project, as it is using Prometheus package registration methods directly:
+    ```go
+    package main
+    
+    import "github.com/prometheus/client_golang/prometheus"
+    
+    func main() {
+        prometheus.Register(prometheus.NewGauge(
+            prometheus.GaugeOpts{
+                Name: "gauge_name",
+                Help: "gauge_help",
+            }))
+    
+        prometheus.MustRegister(prometheus.NewCounter(
+            prometheus.CounterOpts{
+                Name: "counter_name",
+                Help: "counter_help",
+            }))
+    }
+    ```
+
+- The following code is allowed only within `pkg/monitoring`, as it is registering metrics, alerts and recording rules, 
```

Comment:
Not sure you need the whole code here. WDYT about just listing the not-allowed functions?

Reply:
Agreed and updated.